src/edu/hillel/lesson12/Main.java

```diff
@@ -0,0 +1,139 @@
+        //1
+        List<String> strings = new ArrayList<>();
+        strings.add("apple");
+        strings.add("apple");
+        strings.add("apricot");
+        strings.add("lemon");
+        strings.add("apple");
+        strings.add("melon");
+        strings.add("kiwi");
+        strings.add("kiwi");
+        strings.add("citrus");
+        strings.add("avocado");
+        strings.add("avocado");
+        strings.add("watermelon");
+
+        System.out.println("count of kiwi is: "+countOccurrence(strings,"kiwi"));
+        System.out.println("count of apple is: "+countOccurrence(strings,"apple"));
+
+        //2 string
+        String[] stringArray = new String[5];
+        stringArray[0]="apple";
+        stringArray[1]="apricot";
+        stringArray[2]="lemon";
+        stringArray[3]="melon";
+        stringArray[4]="avocado";
+        List<String> stringList = new ArrayList<>();
+        toList(stringArray,stringList);
+        System.out.println("New list is: "+stringList);
+        //2 int
+        Integer[] intArray = new Integer[3];
+        intArray[0]=1;
+        intArray[1]=2;
+        intArray[2]=3;
+        List<Integer> intList = new ArrayList<>();
+        toList(intArray,intList);
+        System.out.println("New list is: "+intList);
+
+        //3
+        List<Integer> intList3 = new ArrayList<>();
+        intList3.add(1);
+        intList3.add(2);
+        intList3.add(3);
+        intList3.add(4);
+        intList3.add(1);
+        intList3.add(1);
+        intList3.add(6);
+        intList3.add(4);
+        intList3.add(3);
+        List<Integer> uniqueList = new ArrayList<>();
+        findUnique(intList3,uniqueList);
+        System.out.println("Unique list is: "+uniqueList);
+
+        //4 - calcOccurrence
+        List<String> stringList4 = new ArrayList<>();
+        stringList4.add("bmw");
+        stringList4.add("volvo");
+        stringList4.add("audi");
+        stringList4.add("volvo");
+        stringList4.add("toyota");
+        stringList4.add("bmw");
+        stringList4.add("toyota");
+        stringList4.add("toyota");
+        stringList4.add("zaz");
+        stringList4.add("toyota");
+
+        calcOccurrence(stringList4);
+
+        //4 findOccurrence
+        List<ItemWithCount> items = findOccurrence(stringList4);
+        for (ItemWithCount item : items) {
+            System.out.println("Item name="+item.getName()+", count of occurrence="+item.getCount());
+        }
+    }
+
+    private static int countOccurrence(List<String> strings, String stringToFind){
+        int cnt=0;
+        for (String string : strings) {
+            if (string.compareTo(stringToFind)==0){
+                cnt++;
+            }
+        }
+        return cnt;
+    }
+
+    private static <T> void toList(T[] inArray,List<T> outList){
+        outList.addAll(Arrays.asList(inArray));
+    }
+
+    private static void findUnique(List<Integer> inList, List<Integer> uniqueList){
```

Comment:
Better to create uniqueList inside of method and return it in the end

Reply:
I have replaced code in both methods and returned collection as return value

Reply:
In SQL you can return something from function as return parameter as input parameter. In second case you can return more than one parameter from function/method. So I decided to return by initialised out of method collection.
I replace code in both methods as you mentioned above

Reply:
Sorry - I have tried to explain why I have done so.

Reply:
Got it. But here better to hide some implementation if we don't want to do something with the list outside of method

Reply:
Ok, thanks. I'll keep this in mind in the future.
But in both cases we create variable outside the method.
In case of using this variable as input params - we can be able to do something with it before the method.
In case of returning as return - no. But maybe in this tasks it's better to return as return. I'll try in next homeworks do so.